sanitise.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tally
+
+import (
+	"bytes"
+)
+
+var (
+	// DefaultReplacementCharacter is the default character used for
+	// replacements.
+	DefaultReplacementCharacter = '_'
+
+	// AlphanumericRange is the range of alphanumeric characters.
+	AlphanumericRange = []SanitiseRange{
+		{rune('a'), rune('z')},
+		{rune('A'), rune('Z')},
+		{rune('0'), rune('9')}}
+
+	// UnderscoreDashCharacters is a slice of underscore, and
+	// dash characters.
+	UnderscoreDashCharacters = []rune{
+		'-',
+		'_'}
+
+	// UnderscoreDashDotCharacters is a slice of underscore,
+	// dash, and dot characters.
+	UnderscoreDashDotCharacters = []rune{
+		'.',
+		'-',
+		'_'}
+
+	// AlphanumericSanitiserOpts is the options to create a sanitiser which uses
+	// the alphanumeric SanitiseFn.
+	AlphanumericSanitiserOpts = SanitiseOptions{
+		NameCharacters: ValidCharacters{
+			Ranges:     AlphanumericRange,
+			Characters: UnderscoreDashCharacters,
```

Comment:
Dunno what makes sense for `statsd`. 

I made `AlphanumericSanitiserOpts` un-exported. We can let each reporter package export it's default sanitiser opts, similar to `m3/sanitise.go` in this PR.